core/kernel/persistence/starsql/helper/RecordProcessor.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2023 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+namespace oat\generis\model\kernel\persistence\starsql\helper;
+
+class RecordProcessor
+{
+    public const LANGUAGE_TAGGED_VALUE_PATTERN = "/^(.*)@([a-zA-Z\\-]{5,6})$/";
+
+    public function filterRecordsByLanguage($entries, $allowedLanguages): array
+    {
+        $filteredValues = [];
+        foreach ($entries as $entry) {
+            // collect all entries with matching language or without language
+            $matchSuccess = preg_match(self::LANGUAGE_TAGGED_VALUE_PATTERN, $entry, $matches);
+            if (!$matchSuccess) {
+                $filteredValues[] = $entry;
+            } elseif (isset($matches[2]) && in_array($matches[2], $allowedLanguages)) {
+                $filteredValues[] = $matches[1];
+            }
+        }
+
+        return $filteredValues;
+    }
+
+    public function filterRecordsByAvailableLanguage($entries, $dataLanguage, $defaultLanguage): array
+    {
+        $fallbackLanguage = '';
+
+        foreach ($entries as $entry) {
```

Comment:
This line throws following error on kitchen instance:
```
2023-12-04 15:35:07 [ERROR] [tao] 'php error(2): foreach() argument must be of type array|object, null given' /tao/code/generis/core/kernel/persistence/starsql/helper/RecordProcessor.php 48

```

Reply:
This have been fixed